### ISSUE TITLE
Terraform plans shows oci_database_autonomous_database will be updated even without any changes to config

### ISSUE THREAD

**Issue by krasappan** [oracle/terraform-provider-oci#1416]
Terraform Version and Provider Version

Terraform v0.15.3
on darwin_amd64

    provider registry.terraform.io/hashicorp/oci v4.35.0

Affected Resource(s)

oci_database_autonomous_database

Terraform Configuration Files

```hcl
module/adw/adw_variables.tf
variable "compartment_ocid" {}
variable "adw_db_name" {}
variable "adw_display_name" {}
variable "adw_admin_password" {}
variable "adw_data_storage_size_in_tbs" {}
variable "adw_cpu_core_count" {}
variable "adw_db_workload" {}
variable "adw_is_auto_scaling_enabled" {}
variable "adw_is_free_tier" {}
variable "adw_license_model" {}
# variable "adw_access_control_enabled" {}
variable "adw_whitelisted_ips" {}

module/adw/adw.tf
resource "oci_database_autonomous_database" "adw" {
  #Required
  admin_password           = var.adw_admin_password
  compartment_id           = var.compartment_ocid
  cpu_core_count           = var.adw_cpu_core_count
  data_storage_size_in_tbs = var.adw_data_storage_size_in_tbs
  db_name                  = var.adw_db_name
  display_name             = var.adw_display_name
  db_workload              = var.adw_db_workload
  is_auto_scaling_enabled  = var.adw_is_auto_scaling_enabled
  is_free_tier             = var.adw_is_free_tier
  license_model            = var.adw_license_model
  # is_access_control_enabled = var.adw_access_control_enabled
  whitelisted_ips           = var.adw_whitelisted_ips
}

main.tf: 

module "adw" {
  source                       = "../modules/adw"
  adw_admin_password           = var.adw_admin_password
  compartment_ocid             = module.compartment.compartment_ocid
  adw_cpu_core_count           = var.adw_cpu_core_count
  adw_data_storage_size_in_tbs = var.adw_data_storage_size_in_tbs
  adw_db_name                  = "SDS${var.env_type}DB"
  adw_display_name             = "SDS-${var.env_type}-${var.env_region}-ADW-DB"
  adw_db_workload              = var.adw_db_workload
  adw_is_auto_scaling_enabled  = var.adw_is_auto_scaling_enabled
  adw_is_free_tier             = var.adw_is_free_tier
  adw_license_model            = var.adw_license_model
  # adw_access_control_enabled   = var.adw_access_control_enabled
  adw_whitelisted_ips = var.adw_whitelisted_ips

}

```
Debug Output

Plan:
```hcl
# module.adw.oci_database_autonomous_database.adw will be updated in-place
  ~ resource "oci_database_autonomous_database" "adw" {
      ~ admin_password                = (sensitive value)
        id                            = "ocid1.autonomousdatabase.oc1.iad.xxxx"
        # (38 unchanged attributes hidden)
    }

```
Apply Output:

```hcl 
Error: 400-InvalidParameter
│ Provider version: 4.35.0, released on 2021-07-14. This provider is 1 update(s) behind to current.
│ Service: Database Autonomous
│ Error Message: Update request for Autonomous Database ocid1.autonomousdatabase.oc1.iad.xxxxx failed. No arguments or arguments same as current configuration were specified in the update request.
│ OPC request ID: 531a32de6ca8716b7b63c9e48d12c58d/30D8D1ACDE82737150AFC2039535D143/B7FCA2CD64B2A70BFD763D4E5C03B047
│ Suggestion: Please update the parameter(s) in the Terraform config as per error message Update request for Autonomous Database ocid1.autonomousdatabase.oc1.iad.xxxxxx failed. No arguments or arguments same as current configuration were specified in the update request.
│
```
Expected Behavior

TF Plan is not expected to show the changes related to ADW. 

Actual Behavior

Though there are no changes in configuration for ADW, terraform plan shows that ADW will be updated during the plan and it 
errors out during apply

Steps to Reproduce

    Terraform plan -Displays ADW to be udpated
    Terraform apply - Error (shared above)

**Comment by varmax2511** [comment]
The update request is detected because there is an update flagged in the admin_password. Once update is done, you are hitting https://github.com/terraform-providers/terraform-provider-oci/issues/1417

**Comment by ravinitp** [comment]
We are very sorry that we couldn't respond to each and every issue reported on GitHub. 
Although we have refined the process to prioritize customer issues on GitHub, since this issue was reported a while ago, there is a good chance it may have been fixed in the latest version of Terraform Provider OCI. 
If you are still experiencing this issue, please create a new issue and label it as Bug.
